src/DonationForms/Blocks/DonationFormBlock/Controllers/BlockRenderController.php

```diff
@@ -22,9 +23,12 @@ class BlockRenderController
      */
     public function render(array $attributes)
     {
-        // return early if we're still inside the editor to avoid server side effects
+        // return early with a mocked button if we're still inside the editor to avoid server side effects
         if (!empty($_REQUEST['post']) || !empty($_REQUEST['action']) || !empty($_REQUEST['_locale'])) {
-            return null;
+            return sprintf(
+                '<button type="button" class="givewp-donation-form-modal__open">%s</button>',
+                $attributes['openFormButton']
+            );
```

Comment:
hmm, will this cause any side effects to the donation form block?  i'm wondering if this should just be handled inside the render files of the donor and donations blocks.

Reply:
Reverted on https://github.com/impress-org/givewp/pull/7812/commits/c4c9db6efe7417b1c4e08189b339e09ccf11587c then improved readability on https://github.com/impress-org/givewp/pull/7812/commits/ec31abfd8c5fc271148939f81045a6f94bb0ff5a

Reply:
Cool, thanks! Looks good to me 😄 

Some additional thoughts that are _not_ meant to be solved here:

Honestly, the more I think about it i'm not sure we even need the return early statement in the donation block render anymore. If I remember correctly, the render_callback had a tendency to fire more times then expected and I was trying to compensate for that.  At some point I want to run some tests without that statement to see if it's necessary anymore.